client/components/premium-global-styles-upgrade-modal/index.tsx

```diff
@@ -30,7 +39,15 @@ export default function PremiumGlobalStylesUpgradeModal( {
 	const translate = useTranslate();
 	const premiumPlanProduct = useSelector( ( state ) => getProductBySlug( state, PLAN_PREMIUM ) );
 	const translations = useGlobalStylesUpgradeTranslations( { numOfSelectedGlobalStyles } );
-	const isLoading = ! premiumPlanProduct;
+	const isPremiumPlanProductLoading = ! premiumPlanProduct;
+	const pricingMeta = usePricingMetaForGridPlans( {
+		planSlugs: [ PLAN_PREMIUM ],
+		storageAddOns: null,
+	} );
+
+	const pricing = pricingMeta?.[ PLAN_PREMIUM ];
+	const isPricingLoaded =
+		pricing?.currencyCode && pricing?.originalPrice.monthly && pricing?.originalPrice.full;
```

Comment:
How do you feel if we had one single `isLoading` for both the plan and its price, so that the whole modal would only appear when everything is ready, rather than risking two separate spinners?

Presumingly, both plan and price should be available at roughly the same time, so IMHO this boils down to simplifying the code rather than improving the UX. 🤔 

Something like this (untested, and not particularly elegant):

```js
const isPremiumPlanProductLoaded = !! premiumPlanProduct;
const isPricingLoaded = pricing?.currencyCode && /* ... */;

const isLoaded = isPremiumPlanProductLoaded && isPricingLoaded;

<Dialog className={ classNames( { loading: ! isLoaded } ) }>
  { ! isLoaded && <LoadingEllipsis /> }
  { isLoaded && /* the whole modal without additional spinners or placeholders */ }
</Dialog>
```

Also note that we have an "isPremiumPlanProduct**Loading**" and an "isPricing**Loaded**". I personally find the difference confusing, and would easily miss it at first glance (I actually only noticed it when I wrote the suggestion above! 🙂 ).
We should probably have them either both `loading` or both `loaded`.

Reply:
> How do you feel if we had one single `isLoading` for both the plan and its price, so that the whole modal would only appear when everything is ready, rather than risking two separate spinners?

Yup, makes sense 👍

> Also note that we have an "isPremiumPlanProduct**Loading**" and an "isPricing**Loaded**". I personally find the difference confusing, and would easily miss it at first glance (I actually only noticed it when I wrote the suggestion above! 🙂 ).
> We should probably have them either both `loading` or both `loaded`.

Definitely, my bad!

Addressed on https://github.com/Automattic/wp-calypso/pull/86057/commits/7a93a5adc4caebd92988e9359a93a17d5cec0660 & https://github.com/Automattic/wp-calypso/pull/86057/commits/a9f95683f5e4b4350df776550d137071726f9341

Reply:
Thanks!
Your changes work fine, but I've realized that there's a fundamental piece missing here!

If you follow all the steps from the beginning, the modal works fine; if you test straight from the Design Picker steps (e.g. `/setup/site-setup/designSetup?siteSlug=SITE_SLUG`), the modal doesn't load anymore.

The issue is caused by the pricing details missing.
If you start from scratch, you would pass through the plans grid, which queries the plans and persists their pricing details (I think in the session cache, but I haven't checked).
If you start from the Design Picker, instead, the pricing details are not queried, and the modal never loads.

The line that fetches plans (and pricing) details in the plans grid is this:

https://github.com/Automattic/wp-calypso/blob/a9f95683f5e4b4350df776550d137071726f9341/client/my-sites/plans-features-main/index.tsx#L707

We should be able to fix our missing prices by adding `<QueryPlans />` to the modal (no need for the `coupons` prop).
Near the other Query component would be a perfect location:

https://github.com/Automattic/wp-calypso/blob/a9f95683f5e4b4350df776550d137071726f9341/client/components/premium-global-styles-upgrade-modal/index.tsx#L64

Reply:
Thanks for debugging this! I've checked it and you're right, adding `<QueryPlans />` to `PremiumGlobalStylesUpgradeModal` makes the pricing available when getting right into the design picker. But I find it confusing that `<QueryPlans />` is needed in order to get the right data out of `usePricingMetaForGridPlans`. Don't you think it should be part of the hook?

Something like:

```
const QueryPlans = ( { coupon }: { coupon?: string } ) => {
	...
};

export const useQueryPlans = ( { coupon }: { coupon?: string } = {} ) => {
	const dispatch = useDispatch();

	useEffect( () => {
		dispatch( requestPlans( coupon ) );
	}, [ dispatch, coupon ] );
};

---

const usePricingMetaForGridPlans: UsePricingMetaForGridPlans = ( { ... }: Props ) => {
	useQueryPlans();
	...
```

But then we'd need to provide the `coupon` prop to `usePricingMetaForGridPlans` as well 🤔

Reply:
`QueryPlans` is a very old component (I've casually seen it on 7 years old commits!), used all over Calypso, and especially on old class components that don't support hooks.

I guess providing some kind of `usePlans` option would be helpful, but given how simple using the component is in any scenario, I can't imagine it being of any priority. 😅

Reply:
Right, I've added the `QueryPlans` component on https://github.com/Automattic/wp-calypso/pull/86057/commits/a9a3e99c0628023d4ec41dd7318589051b04faff

Reply:
@Copons testing steps updated

Reply:
> But I find it confusing that <QueryPlans /> is needed in order to get the right data out of usePricingMetaForGridPlans.

This is no longer the case. `usePricingMetaForGridPlans` works off the data-store query hooks, doesn't rely on Calypso state.

Reply:
`usePricingMetaForGridPlans` is migrating to data-stores itself in another PR (tomorrow or next week).